frontend/view-edit-events/event-home/view-edit-event-materials.component.tsx

```diff
@@ -1,42 +1,168 @@
+import { cloneDeep } from "lodash-es";
 import React from "react";
+import AddEditMaterials from "../../add-leads/add-edit-materials.component";
+import { GrowlType, showGrowl } from "../../growls/growls.component";
 import BasicTableReport from "../../reports/shared/basic-table-report.component";
+import easyFetch from "../../util/easy-fetch";
+import { handlePromiseError } from "../../util/error-helpers";
+import Modal from "../../util/modal.component";
 import { SingleEvent } from "../view-event.component";
+import EventMaterial from "./event-material.component";
 import EventSection from "./event-section.component";
+import { useCss } from "kremling";
 
 export default function ViewEditEventMaterials(
   props: ViewEditEventMaterialsProps
 ) {
+  const scope = useCss(css);
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [materialsDistributed, setMaterialsDistributed] = React.useState([]);
+  const [materials, setMaterials] = React.useState([]);
+  const [showMaterials, setShowMaterials] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!isEditing) {
+      setMaterialsDistributed(cloneDeep(props.event.materialsDistributed));
+    }
+  }, [isEditing, props.event.materialsDistributed]);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    easyFetch(`/api/materials`, { signal: ac.signal })
+      .then(setMaterials)
+      .catch(handlePromiseError);
+    return () => {
+      ac.abort();
+    };
+  }, []);
+
+  React.useEffect(() => {
+    if (isSaving) {
+      const abortController = new AbortController();
+      easyFetch(`/api/events/${props.event.id}`, {
+        method: "PATCH",
+        signal: abortController.signal,
+        body: {
+          materialsDistributed: materialsDistributed,
+        },
+      })
+        .then((updatedEvent) => {
+          props.eventUpdated(updatedEvent);
+          showGrowl({ type: GrowlType.success, message: "Event was updated." });
```

Comment:
This is a nice touch - my guess is you found this pattern in a different file and copied it? I like it because it shows the user once the save finishes.